### PR TITLE
feat: add disabled state for dropdown and table view options

### PR DESCRIPTION
**Pull request by leMaur**
## Summary
https://app.clickup.com/t/pnabvn

This PRs adds `disabled` state for "table view-options" and "dropdown".

## Checklist

-   [x] I checked my UI changes against the design and there are no notable differences
-   [x] I checked my UI changes for any responsiveness issues
-   [x] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I provided a screenshot of my changes to the component _(if applicable)_
-   [ ] I regenerated the `icons.html` file and checked if my newly added icon is shown correctly _(if necessary)_
-   [ ] I added an explanation on how to use the component to the readme _(if necessary)_
-   [x] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [x] Ready to be merged

### CODE REVIEW

File: resources/views/dropdown.blade.php [hunk 1]
```diff
@@ -36,7 +37,9 @@ class="{{ $wrapperClass ? $wrapperClass : 'absolute inline-block top-0 right-0 t
 >
     <div>
         <button
-            @click="{{ $dropdownProperty }} = !{{ $dropdownProperty }}"
+            type="button"
+            @unless($disabled) @click="{{ $dropdownProperty }} = !{{ $dropdownProperty }}" @endunless
+            @if($disabled) disabled @endif
```
Comment (alexbarnsley) [inline resources/views/dropdown.blade.php:42]: could this be an if/else?

```
@unless($disabled)
	@click="{{ $dropdownProperty }} = !{{ $dropdownProperty }}" 
@else
	disabled
@endunless
```

or vice versa

File: resources/views/tables/view-options.blade.php [hunk 2]
```diff
@@ -2,28 +2,33 @@
...
+        type="button"
         :class="{
             '{{ $selectedClasses }}': tableView === 'grid',
             '{{ $unselectedClasses }}': tableView !== 'grid',
         }"
-        class="py-2 px-3 cursor-pointer border-b-3"
-        @click="tableView = 'grid'"
+        class="py-2 px-3 border-b-3"
+        @unless ($disabled) @click="tableView = 'grid'" @endunless
+        @if ($disabled) disabled @endif
```
Comment (alexbarnsley) [inline resources/views/tables/view-options.blade.php:17]: similar to above? 🤔

File: resources/views/tables/view-options.blade.php [hunk 3]
```diff
@@ -2,28 +2,33 @@
...
+        type="button"
         :class="{
             '{{ $selectedClasses }}': tableView === 'list',
             '{{ $unselectedClasses }}': tableView !== 'list',
         }"
-        class="py-2 px-3 cursor-pointer border-b-3"
-        @click="tableView = 'list'"
+        class="py-2 px-3 border-b-3"
+        @unless ($disabled) @click="tableView = 'list'" @endunless
+        @if ($disabled) disabled @endif
```
Comment (alexbarnsley) [inline resources/views/tables/view-options.blade.php:30]: similar to above? 🤔

**Review by alexbarnsley — CHANGES_REQUESTED** [review]

**Review by ItsANameToo — APPROVED** [review]

**Comment by ItsANameToo** [comment]
@leMaur conflict

**Review by ItsANameToo — APPROVED** [review]

**Pull request merged**